src/SFA.DAS.FindEpao/Application/Courses/Queries/GetCourseEpaos/GetCourseEpaosQueryHandler.cs

```diff
@@ -48,6 +49,27 @@ public async Task<GetCourseEpaosResult> Handle(GetCourseEpaosQuery request, Canc
             var filteredEpaos = epaosTask.Result
                 .Where(_courseEpaoIsValidFilterService.IsValidCourseEpao)
                 .ToList();
+
+            IDictionary<string, Task<List<GetStandardsExtendedListItem>>> standardsTasks = 
+                new Dictionary<string, Task<List<GetStandardsExtendedListItem>>>();
+
+            foreach (var filtEpao in filteredEpaos)
+                standardsTasks.Add(filtEpao.EpaoId,_assessorsApiClient.Get<List<GetStandardsExtendedListItem>>(
+                     new GetCourseEpaosStandardVersionsRequest(filtEpao.EpaoId, request.CourseId)));
+            
+            await Task.WhenAll(standardsTasks.Select(x => x.Value));
```

Comment:
You may need to add a .ToArray on here to force the evaluation of the select to properly await the Tasks.

Reply:
I think this overall process can be simplified into
  
```
  var epaoCalls = filteredEpaos.
                    ToDictionary(
                    t => t.EpaoId,
                    t => _assessorsApiClient.Get<IEnumerable<GetStandardsExtendedListItem>>(new GetCourseEpaosStandardVersionsRequest(t.EpaoId, request.CourseId)));
                        
            await Task.WhenAll(epaoCalls.Select(x => x.Value).ToArray());


```            foreach (var filtEpao in filteredEpaos)
            {
                if (epaoCalls.TryGetValue(filtEpao.EpaoId, out var epaoResponse) && epaoResponse.Result != null)
                {
                    var epaoVersions = epaoResponse.Result.Where(c => _courseEpaoIsValidFilterService.ValidateVersionDates(c.EffectiveFrom, c.EffectiveTo))
                            .Select(x => x.Version).ToArray();

                    filtEpao.CourseEpaoDetails.StandardVersions = epaoVersions;
                }
            }
```

Reply:
agreed - tested & updated